### PR TITLE
Added Sudoku

### PR DESCRIPTION
**Pull request by VanshikaMittal07**

### CODE REVIEW

File: Others/Sudoku.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				sms[row / 3][col / 3] |= (1 << arr[row][col]);
+			}
+		}
+		
+		sudoku(arr, 1, rows, cols, sms);
+	}
+	
+	private static void sudoku(int[][] arr, int cellno, int[] rows, int[] cols, int[][] sms){
+		if(cellno > arr.length * arr.length){
+			for(int i = 0; i < arr.length; i++){
```
Comment (yanglbme) [inline Others/Sudoku.java]: `for` loop replaceable with `foreach`.

File: Others/Sudoku.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import java.util.Scanner;
+class Sudoku {
+
+	public static void main(String[] args) {
+		Scanner scn = new Scanner(System.in);
+		
+		int[][] arr = new int[9][9];
+		for(int i = 0; i < arr.length; i++){
```
Comment (yanglbme) [inline Others/Sudoku.java]: Format your code, please. Like this,
```java
for (int i = 0; i < arr.length; ++i) {

}
```

File: Others/Sudoku.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        }
+
+        int row = (cellno - 1) / arr.length;
+        int col = (cellno - 1) % arr.length;
+
+        if (arr[row][col] != 0) {
+            sudoku(arr, cellno + 1, rows, cols, sms);
+        } else {
+            for (int choice = 1; choice <= 9; ++choice) {
```
Comment (yanglbme) [inline Others/Sudoku.java]: Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 9 out of bounds for length 9

**Comment by christianbender** [comment]
@yanglbme I marked the pull request as ***on hold***

**Comment by varunu28** [comment]
@christianbender @yanglbme Why are we reviewing/adding solutions to online judge problems in the repository. The above problem is a typical white-boarding problem. 

Please make sure we add implementations for CS algorithms and data structures and refrain from adding Leetcode/Hackerrank solutions in this repository

@dynamitechetan @AnupKumarPanwar Your views please.

**Comment by yanglbme** [comment]
Hey @varunu28 , I agree with you. Thanks for your reminder :)

**Pull request closed without merging**